clusterloader2/pkg/measurement/common/scheduler_latency.go

```diff
@@ -182,30 +248,7 @@ func (s *schedulerLatencyMeasurement) getSchedulingLatency(c clientset.Interface
 			measurementutil.ConvertSampleToHistogram(sample, preemptionEvaluationHist)
 		}
 	}
-
-	if err := s.setQuantileFromHistogram(&result.E2eSchedulingLatency, e2eSchedulingDurationHist); err != nil {
-		return nil, err
-	}
-	if err := s.setQuantileFromHistogram(&result.SchedulingLatency, schedulingAlgorithmDurationHist); err != nil {
-		return nil, err
-	}
-
-	for _, ePoint := range extentionsPoints {
-		if err := s.setQuantileFromHistogram(result.FrameworkExtensionPointDuration[ePoint], frameworkExtensionPointDurationHist[ePoint]); err != nil {
-			return nil, err
-		}
-	}
-
-	if err := s.setQuantileFromHistogram(&result.PreemptionEvaluationLatency, preemptionEvaluationHist); err != nil {
-		return nil, err
-	}
-
-	content, err := util.PrettyPrintJSON(result)
-	if err != nil {
-		return nil, err
-	}
-	summary := measurement.CreateSummary(schedulerLatencyMetricName, "json", content)
-	return []measurement.Summary{summary}, nil
+	return &schedulerLatencyMetrics{e2eSchedulingDurationHist, schedulingAlgorithmDurationHist, preemptionEvaluationHist, frameworkExtensionPointDurationHist}, nil
```

Comment:
nit:
return &schedulerLatencyMetrics{
  e2eSchedulingDurationHist: ...
  ..
}, nil

This makes it more explicit and less error-prone for extensions.

Reply:
Fixed